core/templates/dev/head/pages/exploration_player/StatsReportingService.js

```diff
@@ -84,6 +99,26 @@ oppia.factory('StatsReportingService', [
         siteAnalyticsService.registerNewCard(1);
 
         stopwatch.reset();
+        totalTime.reset();
+        $rootScope.$broadcast(
```

Comment:
Try to find a way that doesn't use $broadcast, if you can. A $broadcast is a bit like a goto, and we try to avoid it.

Reply:
Ahh. I just learnt about the usage of broadcast (my Angular is still a bit rusty). It seemed like a nice way to keep publishing events and checking whether that event is seen - seemed especially useful for knowing when we crossed our min time for exploration actual start event.